pkg/controller/eventlog/eventlog.go

```diff
@@ -0,0 +1,52 @@
+package eventlog
+
+import (
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/record"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+type ControllerWithEvents interface {
+	EventRecorder() *record.EventRecorder
+	Logger() *logr.Logger
+	Client() *client.Client
+}
+
+type ControllerWithLogger interface {
+	Name() string
+	Logger() *logr.Logger
+}
+
+type ErrorEvent struct {
+	E      error
+	Reason string
+	Owner  *runtime.Object
```

Comment:
`Owner` is never used.

Reply:
I'm also not sure what the advantage of having this struct is?

Reply:
removed